packages/@expo/cli/src/start/server/middleware/ManifestMiddleware.ts

```diff
@@ -360,8 +360,8 @@ export abstract class ManifestMiddleware<
   /** Exposed for testing. */
   async checkBrowserRequestAsync(req: ServerRequest, res: ServerResponse, next: ServerNext) {
     // Read the config
-    const bundlers = getPlatformBundlers(this.initialProjectConfig.exp);
-    if (bundlers.web === 'metro') {
+    const bundlers = getPlatformBundlers(this.projectRoot, this.initialProjectConfig.exp);
+    if (bundlers.web === 'metro' && this.initialProjectConfig.exp.platforms?.includes('web')) {
```

Comment:
This new guard is required as non-web projects still have the `bundler.web === 'metro'`